src/utilities/useHash.ts

```diff
@@ -0,0 +1,46 @@
+'use client'
+
+import { useEffect, useState } from 'react'
+
+export function useHash() {
+  const [hash, setHash] = useState('')
+
+  useEffect(() => {
+    // Set initial hash asynchronously
+    setTimeout(() => {
+      setHash(window.location.hash)
```

Comment:
I think we need a check here to handle `#/archive/forecast ` by redirecting it to `/forecasts/avalanche/archive`. Without this, the bug is fixed, but  if the user copies and shares the URL (`http://nwac.localhost:3000/forecasts/avalanche/olympics#/archive/forecast`) it will take them to the zone rather than the all archived. 

Thoughts since we might be removing this is #315 ?

```suggestion
      const currentHash = window.location.hash
      // Handle '#/archive/forecast'
      if (currentHash === '#/archive/forecast' && window.location.pathname !== '/archive') {
        window.history.replaceState(null, '', `/forecasts/avalanche/archive${currentHash}`)
      }
      setHash(window.location.hash)
```

Reply:
Ah good catch on this. I think the issue is really that the hash isn't getting set to the actual hash on initial render. This causes a route with a set hash (i.e. http://nwac.localhost:3000/forecasts/avalanche/olympics#/forecast/1/165572 to get reset to the initialHash because the useState `hash` gets set to `''` as it's initial state. 

Using   
```
const [hash, setHash] = useState(() =>
    typeof window !== 'undefined' ? window.location.hash : '',
  )
```

Fixes this an allows deep linking to widget views other than the initialHash's widget view (i.e. .../olympics/#olympics). 

The fact that you can deep link to the all forecasts view on a zone page is something that I think we should accept since we're removing the navigation forecast archive link and that we'll eventually be moving away from the widgets. 

It would be a little weird to get a link like `http://nwac.localhost:3000/forecasts/avalanche/olympics#/archive/forecast` and be sort of in a zone in terms of our routing but viewing the archive for all zones. But since there's no back button the user would need to use one of our navigation links to get anywhere else anyways. 

I just don't love the idea of adding a special case here for the forecast archive. But I could be convinced that it's worth it...

Reply:
I'm not crazy about a special case for the forest archive either. Since most of us didn't even know this button existed, I think we can ship it as it without the commit I suggested. If we do an exception for this, we can add it later.